shared/types/indexer_test.go

```diff
@@ -0,0 +1,120 @@
+package types
+
+import (
+	"bytes"
+	"github.com/pokt-network/pocket/shared/crypto"
+	"github.com/stretchr/testify/require"
+	"math/rand"
+	"testing"
+	"time"
+)
+
+func TestTransactionIndexerIndexAndGetters(t *testing.T) {
```

Comment:
I'm a fan of having complex unit tests testing a variety of different complex cases, but since this is a standalone package that we likely won't revisit in the future (unless there's a bug or new feature), can you please add behaviour specific unit tests

Here is a non-exhaustive list just as a starting point/reference: 
```go
func TestTransactionIndexerIndexByHeight(t * testing.T)
func TestTransactionIndexerIndexByHash(t * testing.T)
func TestTransactionIndexerIndexBySender(t * testing.T)
func TestTransactionIndexerIndexByRecipient(t * testing.T)
...

func TestTransactionIndexerIndexByHeightOnly(t * testing.T)
func TestTransactionIndexerIndexBySenderOnly(t * testing.T)
func TestTransactionIndexerIndexByRecipientOnly(t * testing.T)
...
```

Reply:
I don't understand. I followed the same testing framework like the Consensus Module where I test public functions covering happy/bad path use cases.

I need more detail on this before I arbitrarily start increasing the number of tests when all of these are already covered

Reply:
> I followed the same testing framework like the Consensus Module where I test public functions covering happy/bad path use cases

I don't think I did a good job on the consensus module level tests when we first implemented it, and definitely think we should add finer grained ones as well, ina addition to the module-level ones.

> I need more detail on this before I arbitrarily start increasing the number of tests when all of these are already covered

I was thinking something similar to what we have for the actor level tests in persistence and utility. Example:

We have unit tests like so:

```
TestInsertFishermanAndExists
TestUpdateFisherman
TestGetFishermenReadyToUnstake
TestGetFishermanStatus
TestGetFishermanPauseHeightIfExists
TestSetFishermanPauseHeightAndUnstakeLater
```

In addition to end-to-end tests like so:

```
FuzzFisherman
```

I think it's important that we do this now and have them in place before merging this in.